lalrpop/src/grammar/parse_tree.rs

```diff
@@ -375,11 +377,22 @@ pub struct MacroSymbol {
 
 impl TerminalString {
     pub fn quoted(i: InternedString) -> TerminalString {
-        TerminalString::Literal(TerminalLiteral::Quoted(i))
+        TerminalString::Literal(TerminalLiteral::Quoted(i, 1))
     }
 
     pub fn regex(i: InternedString) -> TerminalString {
-        TerminalString::Literal(TerminalLiteral::Regex(i))
+        TerminalString::Literal(TerminalLiteral::Regex(i, 0))
+    }
+
+    pub fn with_match_precedence(self, p: usize) -> TerminalString {
+        let base_precedence = p * 2;  // Multiply times two since we still want to distinguish between quoted and regex precedence
```

Comment:
I hadn't realized that this would be necessary, but I guess with `_` it makes sense -- i.e., within a section, quoted must take precedence still (but not across sections). Very good.